### PR TITLE
ci: Update CI to barebone python [all tests ci]

### PR DESCRIPTION
**Pull request by lsetiawan**
This PR is a follow up to https://github.com/OSOceanAcoustics/echopype/pull/1191 since even new PR's are failing!

Conda is now removed from CI since there isn't any special packages that we need from there anymore and we're basically already installing via pip through conda anyways. Plus this was a recommendation from Filipe to do over the summer.

### CODE REVIEW

File: echopype/tests/convert/test_parsed_to_zarr.py [hunk 1]
```diff
@@ -15,6 +15,8 @@
 from echopype.echodata.convention import sonarnetcdf_1
 from echopype.convert.api import _check_file, SONAR_MODELS
 
+pytestmark = pytest.mark.skip(reason="Removed Parsed2Zarr")
```
Comment (emiliom) [inline echopype/tests/convert/test_parsed_to_zarr.py:18]: Is this relevant to this PR?

**Comment by emiliom** [comment]
Looks good, thanks! I made one small inline comment. My only other comment is that it looks like the files `.ci_helpers/py${{ matrix.python-version }}.yaml` are no longer used; or are they actually used in other actions? If not, let's delete them, right?

File: echopype/tests/convert/test_parsed_to_zarr.py (hunk 1 above)
Comment (lsetiawan) [reply echopype/tests/convert/test_parsed_to_zarr.py:18]: > Is this relevant to this PR?

Yepp, without the skip, it'll have a botocore error because it's spinning up moto server multiple times... I have to figure that out in PR https://github.com/OSOceanAcoustics/echopype/pull/1185. For now, it's best to skip this test since lots of changes a coming!

**Comment by lsetiawan** [comment]
> re they actually used in other actions?

Looks like they're still being used in the `windows-utils-test` shenanigans that I'm not sure what it's doing. Probably good to have a tests clean up soon that include this stuff.

File: echopype/tests/convert/test_parsed_to_zarr.py (hunk 1 above)
Comment (emiliom) [reply echopype/tests/convert/test_parsed_to_zarr.py:18]: Got it

**Comment by emiliom** [comment]
> Looks like they're still being used in the windows-utils-test shenanigans that I'm not sure what it's doing. Probably good to have a tests clean up soon that include this stuff.

Sounds good. Merge away, and thanks!!

**Review by emiliom — APPROVED** [review]
